### ISSUE TITLE
WS03: Pass logits directly to CrossEntropyLoss (Pytorch already applies softmax)

### ISSUE THREAD

**Issue by SaifSyed08** [Graylab/DL4Proteins-notebooks#13]
The CNN applies softmax before passing its output to `CrossEntropyLoss`:

```python
output = F.softmax(logits, dim=1)
return output
```

`CrossEntropyLoss` already applies log-softmax internally, so the model should return raw logits:

```python
return logits
```
